docs/how-to-guides/how-to-determine-component-dependencies.md

```diff
@@ -0,0 +1,78 @@
+# How to determine component dependencies
+
+For any developers who wish to try and deploy Autoware as a microservices architecture, it is necessary to understand the software dependencies, communication and implemented features of each ROS package/node.
+
+As an example, the commands necessary to determine the dependencies for the Perception component are shown below.
+
+## Perception component dependencies
+
+To generate a graph of package dependencies, use the following `colcon` command:
+
+```bash
+colcon graph --dot --packages-up-to tier4_perception_launch | dot -Tpng -o graph.png
+```
+
+<!-- TODO add image link here once added to the repository -->
+
+To generate a list of dependencies, use:
+
+```bash
+colcon list --packages-up-to tier4_perception_launch
+```
+
+??? colcon list output
```

Comment:
Whoops. Since the list of nodes was long, I thought that it would be better to hide them in a [collapsible block](https://squidfunk.github.io/mkdocs-material/reference/admonitions/#collapsible-blocks). 

However, I forgot the blank line after the `???` and then the autofix commit removed the indent from lines 24-69.  

Is a collapsible block better, or is it okay to just list out all of the nodes [as per your other suggestion](https://github.com/autowarefoundation/autoware-documentation/pull/162/files#r909135928)?

Reply:
> Whoops. Since the list of nodes was long, I thought that it would be better to hide them in a [collapsible block](https://squidfunk.github.io/mkdocs-material/reference/admonitions/#collapsible-blocks).

I see, it's okay. I'll leave it to you!

Reply:
I noted that the suggested list of topics was different to the original list. Is the list that you suggested the actual output of running `colcon list --packages-up-to tier4_perception_launch`? I don't know myself since I just copied the list from https://github.com/autowarefoundation/autoware-projects/issues/34. 😅 

If the suggestion is the actual output, then I think it would be better to add that instead so that it matches whatever a developer would see when they run that command (assuming no nodes are added or removed in the meantime).

Reply:
> Is the list that you suggested the actual output of running colcon list --packages-up-to tier4_perception_launch?

Yes. If you generate the output you copied, you have to add the `--names-only` option.

```console
$ colcon list --packages-up-to tier4_perception_launch --names-only
ament_cmake_auto
autoware_auto_geometry_msgs
autoware_auto_mapping_msgs
autoware_auto_perception_msgs
autoware_auto_planning_msgs
autoware_auto_vehicle_msgs
autoware_cmake
autoware_lint_common
autoware_point_types
compare_map_segmentation
euclidean_cluster
grid_map_cmake_helpers
grid_map_core
grid_map_cv
grid_map_msgs
grid_map_pcl
grid_map_ros
ground_segmentation
image_projection_based_fusion
image_transport_decompressor
interpolation
kalman_filter
lanelet2_extension
lidar_apollo_instance_segmentation
map_based_prediction
multi_object_tracker
mussp
nav2_common
nav2_msgs
object_merger
object_range_splitter
occupancy_grid_map_outlier_filter
pointcloud_preprocessor
pointcloud_to_laserscan
shape_estimation
tensorrt_yolo
tier4_autoware_utils
tier4_debug_msgs
tier4_pcl_extensions
tier4_perception_launch
tier4_perception_msgs
traffic_light_classifier
traffic_light_map_based_detector
traffic_light_ssd_fine_detector
traffic_light_visualization
vehicle_info_util
```